### ISSUE TITLE
RxMoya linker error for watchOS via project & carthage (Xcode 9.3.1)

### ISSUE THREAD

**Issue by simonrice** [Moya/Moya#1669]
When building RxMoya via Carthage and in the Xcode project with "Generic watchOS device", I get a linker error.  The version in question is Moya 11.0.2 in Xcode 9.3.1, and I've also attempted and failed to build with current `master` too.  From playing around with the project, it also affects ReactiveMoya.

The full linker error is:

```
Undefined symbols for architecture armv7k:
  "_OBJC_CLASS_$_UIImage", referenced from:
      objc-class-ref in SignalProducer+Response.o
ld: symbol(s) not found for architecture armv7k
```

I do have 2x separate workarounds as things stand, neither of which I feel viable for contributing to the codebase:

1.  Adding `Image.swift` to the RxMoya target (as well as Moya).
2. Adding these imports to the top of `Observable+Response.swift` and  `Single+Response.swift`:

```swift
#if os(watchOS)
    import UIKit
#endif
```

The impression I get is tvOS and iOS automatically brings in `UIKit.framework`, whereas watchOS doesn't unless you explicitly import it in your target.

**Comment by sunshinejr** [comment]
Hey @simonrice, thank you so much for a very detailed issue! 💯 

So this is something I noticed as well when I started migrating to Xcode 9.3 in #1656. I actually did what you also suggested in second point, importing `UIKit.UIImage` where possible. I think it fixed the original problem you encountered but I've got stuck on some other error that happened after the fix, probably now connected more to the Carthage than Moya, but still I didn't finish the job, unfortunately.

If you have some experience in Carthage and want to help out in finishing the PR, please hesitate to do so 😉

**Comment by simonrice** [comment]
Fixed by #1656.

**Comment by sunshinejr** [comment]
Oh, thanks for closing it @simonrice, totally forgot about the issue 😄 We still need to prepare a new major version to get this through, but in the meantime you can use `development` branch.
